ml_metadata/tools/mlmd_bench/fill_nodes_workload.cc

```diff
@@ -0,0 +1,321 @@
+         std::is_same<NT, Execution>::value ||
+         std::is_same<NT, Context>::value))
+      << "Unexpected Node Types";
+  // Insert nodes cases.
+  // Loops over all the node inside `node_batch` and sets up one by one.
+  for (int64 i = 0; i < node_batch.size(); ++i) {
+    node_batch[i].set_name(absl::StrCat(node_batch_name, "_node_", i));
+    node_batch[i].set_type_id(type.id());
+    // Uses "********" as the fake property value for current node.
+    std::string property_value(string_value_bytes, '*');
+    int64 curr_num_properties = 0;
+    // Loops over the types properties while generating the node's properties
+    // accordingly.
+    auto it = type.properties().begin();
+    while (curr_num_properties < num_properties &&
+           it != type.properties().end()) {
+      (*node_batch[i].mutable_properties())[it->first].set_string_value(
+          property_value);
+      curr_num_properties++;
+      it++;
+    }
+    // If the node's number of properties is greater than the type(the iterator
+    // of the type properties has reached the end, but `curr_num_properties`
+    // is still less than `num_properties`), uses custom properties instead.
+    while (curr_num_properties < num_properties) {
+      (*node_batch[i]
+            .mutable_custom_properties())[absl::StrCat("custom_p-",
+                                                       curr_num_properties)]
+          .set_string_value(property_value);
+      curr_num_properties++;
+    }
+    curr_bytes += GetTransferredBytes<NT>(node_batch[i]);
+  }
+}
+
+// Sets additional fields(uri, state) for artifacts and returns transferred
+// bytes for these additional fields.
+int64 SetArtifactsAdditionalFields(
+    const std::string& node_batch_name,
+    google::protobuf::RepeatedPtrField<Artifact>& node_batch) {
+  int64 bytes = 0;
+  for (int64 i = 0; i < node_batch.size(); ++i) {
+    node_batch[i].set_uri(absl::StrCat(node_batch_name, "_node_", i, "_uri"));
+    node_batch[i].set_state(Artifact::UNKNOWN);
+    bytes += node_batch[i].uri().size() + 1;
+  }
+  return bytes;
+}
+
+// Sets additional field(state) for executions and returns transferred
+// bytes for these additional fields.
+int64 SetExecutionsAdditionalFields(
+    google::protobuf::RepeatedPtrField<Execution>& node_batch) {
+  int64 bytes = 0;
+  for (int64 i = 0; i < node_batch.size(); ++i) {
+    node_batch[i].set_last_known_state(Execution::UNKNOWN);
+    bytes += 1;
+  }
+  return bytes;
+}
+
+}  // namespace
+
+FillNodes::FillNodes(const FillNodesConfig& fill_nodes_config,
+                     const int64 num_operations)
+    : fill_nodes_config_(fill_nodes_config), num_operations_(num_operations) {
+  switch (fill_nodes_config_.specification()) {
+    case FillNodesConfig::ARTIFACT: {
+      name_ = "fill_artifact";
+      break;
+    }
+    case FillNodesConfig::EXECUTION: {
+      name_ = "fill_execution";
+      break;
+    }
+    case FillNodesConfig::CONTEXT: {
```

Comment:
qq, fill_nodes_config_.specification().DebugString() ?

absl::StrCat("fill_", fill_nodes_config_.specification().DebugString() )

enum string value functions [1]

[1] https://developers.google.com/protocol-buffers/docs/reference/cpp-generated#enum

Reply:
DebugString() can only used for message interface.